lib/iris/fileformats/netcdf.py

```diff
@@ -266,6 +266,10 @@ def ndim(self):
         return len(self.shape)
 
     def __getitem__(self, keys):
+        try:
+            expected_shape = tuple(map(len, keys))
```

Comment:
One key feature of biggus is the ability to compute shapes of arrays after indexing.

You might get better mileage using that:

```
>>> full_shape = (3, 4, 12)
>>> keys = (slice(1, None, 2), (1, 2, 1), 0)
>>> print biggus.zeros(full_shape).__getitem__(keys=keys).shape
(1, 3)
```